specs/dwritecore/DWriteCore.md

```diff
@@ -0,0 +1,255 @@
+as well as mechanisms to avoid the use of compiler extensions like __declspec(uuid(x)) and __uuidof.
+
+Another consideration on non-Windows platforms is the definition of WCHAR. Most projects define WCHAR as 
+wchar_t which is a 32-bit type everywhere except Windows, but Office relies on WCHAR being 16 bits. 
+Therefore, DWriteCore’s cross-platform headers defines WCHAR as either char16_t or wchar_t, depending on 
+a preprocessor definition, and we build both 16-bit and 32-bit flavors.
+
+This is outside the scope of the API itself, but is mentioned here because other cross-platform libraries 
+will have to solve the same problem, and if each library solves the problem in its own way, we may end up
+with interoperability issues.
+
+# Description
+
+New APIs introduced by DWriteCore include:
+
+ - A new DWRITE_FACTORY_TYPE enumeration for creating a _restricted factory_ object
+ - A new method for getting pixel data from a bitmap render target
+
+The new DWRITE_FACTORY_TYPE_RESTRICTED enum value complements the existing values for creating a _shared factory_
+and _isolated factory_ objects. A restricted factory is more locked-down than either of the existing factory
+types. First, it only caches data in-process, and will not attempt to either read from or write to a cross-process
+font cache or persisted cache file. Second, a restricted factory only enumerates well-known fonts in the system
+font collection. The well-known-fonts restriction enables DWriteCore to efficiently construct the system font 
+collection without depending on a cross-process or persistent cache. An example use case for a restricted factory
+would be a web browser rendering process with very limited permissions.
+
+A _bitmap render target_ (IDWriteBitmapRenderTarget) is an API object that encapsulates a system memory bitmap
+and enables rendering glyphs to the bitmap. On Windows, the object encapsulates a GDI memory DC with a GDI
+device-dependent bitmap (DIB) selected into it. However, on non-Windows platforms, there is no GDI DC,
+and the bitmap is just a system memory array. DWriteCore introduces a new version of the bitmap render target
+interface that an application can use to access the bitmap data without going through GDI.
+
+# Examples
+
+## Creating a restricted factory
+
+An application calls the [DWriteCreateFactory](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-dwritecreatefactory)
+function to create a factory object, which is the entry-point to all other DirectWrite APIs. An application can
+create a _restricted factory_ if it does not want the factory or objects created from it to read from or write
+to any cross-process or persistent cache. In addition, a restricted factory's system font collection only includes
+well-known system fonts.
+
+The following example creates a restricted factory.
+
+```c++
+ComPtr<IDWriteFactory7> spFactory;
+HRESULT hr = DWriteCreateFactory(
+    DWRITE_FACTORY_TYPE_RESTRICTED,
```

Comment:
The term "restricted" is ambiguous because it could mean

1. Access to this factory is restricted. Only super-powerful apps get access to it. (Compare "restricted capabilities" which are capabilities that only super-powerful apps get access to.)
2. This factory's actions are restricted. Even super-weak apps can use it.

This is a restriction of type 2. Maybe we can use a less ambiguous term.

The term "isolated" is already being used to mean "no cross-process shared state or persistent cache". This is an extension of "isolated" to limit the font enumeration, so maybe call it "isolated2"?

I think "isolated2" is better than trying to find another adjective. Most of the adjectives are just synonyms of each other, so the distinction is meaningless. Like, what's the difference between "restricted', "isolated", or "constrained"?

Reply:
DISCUSSION: This is a new enum, the existing values are "shared" and "isolated" ... shared produces singleton dwrite instances, isolated means you get your very own (but uses cross-process caching.)  This "restricted" value is both isolated (not shared) and limited to in-process content and a subset of available system fonts.  The app is in control of what the instance gets.

In-box DWrite works with the font caching service to let low-privileged apps still get the performance benefits of cross-process sharing.  Some caching goes into %TEMP% which is a per-application cache location.

RECOMMEND: `_ISOLATED2` seems not a bad answer, but needs clear documentation.

Reply:
DISCUSSION: There's a cost for non-caching (memory, cpu, pages.) Have a clear recommendation to use the `_SHARED` model as much as possible for maximal caching.  The `_ISOLATED2` mode is sharper, and puts the onus on the app to use it only when appropriate (like controlling the list of available fonts.)

Reply:
Renamed and made the recommendation to use _SHARED more prominent in the comments.